### PR TITLE
Feature/favorites

### LINKED ISSUE #20: Add favorite functionality
User Story - As a user, I can favorite and unfavorite breweries. I can see my favorite breweries in a favorites tab. I can also filter my search results by my favorites.

- Additionally, add testing for this implementation.

### PR DESCRIPTION
**Pull request by reneepinna**
What I did:
- Create a favorite context with a custom provider and hook
- Allow the favorite context to add and remove favorited breweries
- Allow the favorite context to track if the favorite local breweries filter is on and create a list of breweries to be displayed
- Connect filter functionality to Map and Brewery container
- Add a button to favorite/unfavorite a brewery
- Created a navLink to the favorites page 
- Created a favorites page that holds all favorite breweries

Additional Changes:
- Moved the header into the homepage
- Formatted the phone number for brewery cards

Did this break anything?

- [x] No
- [ ]  Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x]  Styling 
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [x]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [ ]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next?
- The favorite filter button should be refactored/styled for clarity so that the user knows the options and what they are on
- There should be a nav bar so that users can move between favorites and the main page
- There are some opportunities for refactor

closes #20

### CODE REVIEW

File: src/Components/BreweryCard/BreweryCard.jsx [hunk 1]
```diff
@@ -1,29 +1,75 @@
...
+    }
+  }, []);
+
+  function toggleIsFavorite() {
+    if (isFavorite) {
+      setIsFavorite(false);
+    } else {
+      setIsFavorite(true);
+    }
+  }
```
Comment (banjaminh) [inline src/Components/BreweryCard/BreweryCard.jsx:26]: Looks great! Love the simple implementation.

**Review by Jwensloff — COMMENTED** [review]
This looks great!

File: src/Components/BreweryCard/BreweryCard.jsx [hunk 2]
```diff
@@ -1,29 +1,75 @@
...
+          rel='noreferrer'
+        >
+          Directions
+        </a>}
+      </div>
+      <button
+        className='breweryCard-favorites-btn'
+        onClick={() => {
+          toggleFavorite(brewery);
+          toggleIsFavorite();
```
Comment (hfaerber) [inline src/Components/BreweryCard/BreweryCard.jsx:59]: From a 'speaking to your code' perspective, how would you describe the difference between `toggleFavorite` and `toggleIsFavorite` to a classmate who hasn't ever played with the contextApi.  How would you explain using/needing both?

I'd love to hear some reflection on this in a DM in our slack message!

File: src/Components/BreweryContainer/BreweryContainer.jsx [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+    });
+  }
+
+  useEffect(() => {
+    setCards(createCards(getFilteredBreweries()))
+  }, [favorites, breweries, favoriteFilter])
+
+  const styles = {
+    backgroundColor: favoriteFilter ? '#A9721F' : '#e0cc99'
+  }
```
Comment (hfaerber) [inline src/Components/BreweryContainer/BreweryContainer.jsx:26]: I love how cleanly you've broken this out here!

File: src/Components/Map/Map.jsx [hunk 4]
```diff
@@ -83,7 +83,7 @@ function Map() {
       3958 *
       Math.acos(
         Math.sin(latRad1) * Math.sin(latRad2) +
-          Math.cos(latRad1) * Math.cos(latRad2) * Math.cos(longRad2 - longRad1)
+          Math.cos(latRad1) * Math.cos(latRad2) * Math.cos(longRad2 - longRad1),
```
Comment (hfaerber) [inline src/Components/Map/Map.jsx:86]: Y'all are mathing some math here - nice work!

**Review by hfaerber — COMMENTED** [review]
Nice work on this implementation!  I'm happy to see that you're learning a lot and able to start applying that learning!

**Pull request merged**